pkg/new_storage/general.go

```diff
@@ -403,15 +403,11 @@ func NewBackupDestination(cfg *config.Config) (*BackupDestination, error) {
 			cfg.General.DisableProgressBar,
 		}, nil
 	case "s3":
-		partSize := cfg.General.MaxFileSize / 10000
```

Comment:
These changes limit maximum upload size to 50Gb for original S3
let's adjust your PR

* setup default configuration `PartSize` to zero it mean no restriction to part size
* return code which calculate partSize 
* add 
```go
if cfg.S3.PartSize > 0 && partSize > cfg.S3.PartSize { 
  partSize = cfg.S3.PartSize
}
```

Reply:
Maybe we also should add max limits to 5Gb for calculated `partSize`

Reply:
https://github.com/AlexAkulov/clickhouse-backup/blob/0bc5d7c5aaef304aea09d429a7b853940cf0c825/pkg/backup/upload.go#L242
cfg.General.MaxFileSize config parameter limits the "preferred" size of uploaded files, but shouldn't limit the upload process at all, as I see it.

So, if we have files in our backup that are larger than MaxFileSize, should we abort the backup process altogether? I think no. I don't want to increase MaxFileSize to high values because that will create big parts to upload.
If we will have 2 independent settings for the preferred size of the uploaded file and for setting the size of the chunk, we will get a more flexible solution - we will be able to set both the preferred size of the uploaded file and the maximum size of the file uploaded to S3. I agree to set PartSize to 0 by default and calculate the default value in this case.

I've already updated my PR to described logic.

Reply:
`cfg.General.MaxFileSize` have not clear mental name, and not
it limits archive size on remote storage, and we split data parts by this parameter, to avoid too big archive sizes
if we have one part more than `cfg.General.MaxFileSize` it still tries to upload, but just in the separate `.tar` file on remote storage

Reply:
> I don't want to increase MaxFileSize to high values because that will create big parts to upload.

You don't need to increase the `MaxFileSize` which related to max size of archive on remote storage (one table -> one and more archive which depends on disks mapping and table size)

you should increase `partSize`, big `partSize` for multipart S3 upload increase upload speed on S3, 
if you agree  with my proposal, you can limit max `partSize` for data part during s3 multipart upload to `Minio`

I ask again, did you change anything on `minio` side? By default, it allows the same limits for multipart upload as original S3.

In real cases, if you have `0.5TB+` table you can have 50Gb+ data parts in one table

Reply:
No, I I haven't changed anything in MinIO settings.
As we can see from the [S3 dcos](https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html) they have the same 10,000 parts upload limit. If we know this limitation for S3, it would be nice to display an error and not try to upload files for which: **file_size/part_size>10000**.
At the moment, you can only find out about this by enabling the S3_DEBUG setting.